files/scripts/remove_db_replication.sh

```diff
@@ -6,3 +6,4 @@ INSTANCE_ID=`curl http://instance-data/latest/meta-data/instance-id`
 SUBSCRIPTION=${USER}_${INSTANCE_ID:2}
 
 psql -U $USER -c "DROP SUBSCRIPTION $SUBSCRIPTION;"
+touch /var/www/wcivf/home/server_dirty
```

Comment:
To move to an issue / consider.

If there's a problem with replication, the server will boot with a `server_dirty` file, delete it and then re-make it.

This might make it hard to debug if instances start cycling out and we have to log in to look at them.

It might help to have a log that we can write to in these scripts and a nice way of reading that log, to see what's actually going on when initially debugging.

Reply:
> If there's a problem with replication, the server will boot with a server_dirty file, delete it and then re-make it.

So I think what would happen would be that if something goes wrong in the `setup_db_replication` script, as the `rm /var/www/wcivf/home/server_dirty` is the final step, the `server_dirty` file remains. Then when the instance gets cycled out, since the service failed, the `ExecStop=remove_db_replication.sh` exit command won't run.

As for logging, would definitely be good to have a way to log from the bash scripts - I'll have a look in to this. What I have been using to debug so far is `jounalctl` that lets you get logs for a service e.g.:
```
root@ip-172-31-29-109:/home/michael# journalctl -u replication.service | tail -n 50
Jul 06 16:27:18 ip-172-31-29-109 setup_db_replication.sh[1340]:   Applying news_mentions.0001_initial... OK
Jul 06 16:27:18 ip-172-31-29-109 setup_db_replication.sh[1340]:   Applying news_mentions.0002_ballotnewsarticle_publisher... OK
Jul 06 16:27:18 ip-172-31-29-109 setup_db_replication.sh[1340]:   Applying news_mentions.0003_auto_20210225_1103... OK
Jul 06 16:27:18 ip-172-31-29-109 setup_db_replication.sh[1340]:   Applying parishes.0001_initial... OK
Jul 06 16:27:18 ip-172-31-29-109 setup_db_replication.sh[1340]:   Applying parties.0003_auto_20160422_1148... OK
Jul 06 16:27:18 ip-172-31-29-109 setup_db_replication.sh[1340]:   Applying parties.0004_auto_20170530_1520... OK
Jul 06 16:27:18 ip-172-31-29-109 setup_db_replication.sh[1340]:   Applying parties.0005_auto_20170530_1736... OK
Jul 06 16:27:18 ip-172-31-29-109 setup_db_replication.sh[1340]:   Applying parties.0006_auto_20180417_1711... OK
Jul 06 16:27:18 ip-172-31-29-109 setup_db_replication.sh[1340]:   Applying parties.0007_auto_20180417_1733... OK
Jul 06 16:27:18 ip-172-31-29-109 setup_db_replication.sh[1340]:   Applying parties.0008_auto_20180422_1248... OK
Jul 06 16:27:18 ip-172-31-29-109 setup_db_replication.sh[1340]:   Applying parties.0009_manifesto_easy_read_url... OK
Jul 06 16:27:18 ip-172-31-29-109 setup_db_replication.sh[1340]:   Applying parties.0010_localparty_is_local... OK
Jul 06 16:27:18 ip-172-31-29-109 setup_db_replication.sh[1340]:   Applying people.0020_person_favourite_biscuit... OK
Jul 06 16:27:18 ip-172-31-29-109 setup_db_replication.sh[1340]:   Applying people.0021_person_twfy_id... OK
Jul 06 16:27:18 ip-172-31-29-109 setup_db_replication.sh[1340]:   Applying people.0022_auto_20170605_1425... OK
Jul 06 16:27:18 ip-172-31-29-109 setup_db_replication.sh[1340]:   Applying people.0023_auto_20170605_1559... OK
Jul 06 16:27:18 ip-172-31-29-109 setup_db_replication.sh[1340]:   Applying people.0024_personpost_elected... OK
Jul 06 16:27:18 ip-172-31-29-109 setup_db_replication.sh[1340]:   Applying people.0025_personpost_post_election... OK
Jul 06 16:27:18 ip-172-31-29-109 setup_db_replication.sh[1340]:   Applying people.0026_auto_20180417_1944... OK
Jul 06 16:27:18 ip-172-31-29-109 setup_db_replication.sh[1340]:   Applying pledges.0001_initial... OK
Jul 06 16:27:18 ip-172-31-29-109 setup_db_replication.sh[1340]:   Applying pledges.0002_auto_20180502_0946... OK
Jul 06 16:27:18 ip-172-31-29-109 setup_db_replication.sh[1340]:   Applying peoplecvs.0001_initial... OK
Jul 06 16:27:18 ip-172-31-29-109 setup_db_replication.sh[1340]:   Applying peoplecvs.0002_auto_20170522_1324... OK
Jul 06 16:27:18 ip-172-31-29-109 setup_db_replication.sh[1340]:   Applying people.0027_person_party_ppc_page_url... OK
Jul 06 16:27:18 ip-172-31-29-109 setup_db_replication.sh[1340]:   Applying people.0028_person_pk_to_int... OK
Jul 06 16:27:18 ip-172-31-29-109 setup_db_replication.sh[1340]:   Applying people.0029_person_last_updated... OK
Jul 06 16:27:18 ip-172-31-29-109 setup_db_replication.sh[1340]:   Applying people.0030_insta-youtube... OK
Jul 06 16:27:18 ip-172-31-29-109 setup_db_replication.sh[1340]:   Applying people.0031_person_sort_name... OK
Jul 06 16:27:18 ip-172-31-29-109 setup_db_replication.sh[1340]:   Applying people.0032_person_death_date... OK
Jul 06 16:27:18 ip-172-31-29-109 setup_db_replication.sh[1340]:   Applying people.0033_facebookadvert... OK
Jul 06 16:27:18 ip-172-31-29-109 setup_db_replication.sh[1340]:   Applying people.0034_auto_20191213_0942... OK
Jul 06 16:27:18 ip-172-31-29-109 setup_db_replication.sh[1340]:   Applying people.0035_personpost_votes_cast... OK
Jul 06 16:27:18 ip-172-31-29-109 setup_db_replication.sh[1340]:   Applying people.0036_auto_20191213_1028... OK
Jul 06 16:27:18 ip-172-31-29-109 setup_db_replication.sh[1340]:   Applying people.0037_auto_20210604_1245... OK
Jul 06 16:27:18 ip-172-31-29-109 setup_db_replication.sh[1340]:   Applying people.0038_add_party_name_to_personposts... OK
Jul 06 16:27:18 ip-172-31-29-109 setup_db_replication.sh[1340]:   Applying peoplecvs.0003_delete_cvs... OK
Jul 06 16:27:18 ip-172-31-29-109 setup_db_replication.sh[1340]:   Applying profiles.0001_initial... OK
Jul 06 16:27:18 ip-172-31-29-109 setup_db_replication.sh[1340]:   Applying referendums.0001_initial... OK
Jul 06 16:27:18 ip-172-31-29-109 setup_db_replication.sh[1340]:   Applying results.0001_initial... OK
Jul 06 16:27:18 ip-172-31-29-109 setup_db_replication.sh[1340]:   Applying results.0002_auto_20170607_1546... OK
Jul 06 16:27:18 ip-172-31-29-109 setup_db_replication.sh[1340]:   Applying results.0003_personpostresult... OK
Jul 06 16:27:18 ip-172-31-29-109 setup_db_replication.sh[1340]:   Applying results.0004_person_post_p_key... OK
Jul 06 16:27:18 ip-172-31-29-109 setup_db_replication.sh[1340]:   Applying robots.0001_initial... OK
Jul 06 16:27:18 ip-172-31-29-109 setup_db_replication.sh[1340]:   Applying sessions.0001_initial... OK
Jul 06 16:27:18 ip-172-31-29-109 setup_db_replication.sh[1340]:   Applying sites.0002_alter_domain_unique... OK
Jul 06 16:27:18 ip-172-31-29-109 setup_db_replication.sh[1340]: + psql wcivf -U wcivf -c 'CREATE SUBSCRIPTION wcivf_0938517d091a04218 CONNECTION '\''dbname=wcivf host=wcivf.cvidnzxjxvjf.eu-west-2.rds.amazonaws.com user=wcivf password=C4MufxN4UnCTc6'\'' PUBLICATION alltables;'
Jul 06 16:27:19 ip-172-31-29-109 setup_db_replication.sh[1340]: NOTICE:  created replication slot "wcivf_0938517d091a04218" on publisher
Jul 06 16:27:19 ip-172-31-29-109 setup_db_replication.sh[1340]: CREATE SUBSCRIPTION
Jul 06 16:27:19 ip-172-31-29-109 setup_db_replication.sh[1340]: + rm /var/www/wcivf/home/server_dirty
Jul 06 16:27:19 ip-172-31-29-109 systemd[1]: Started Setup DB replication.
```

This helped me debug what was going wrong the service `ExecStart` and `ExecStop` commands